### DISCUSSION TITLE
Another Turtle Question

### DISCUSSION THREAD

**Discussion by SirEndii** [cc-tweaked/CC-Tweaked#741] [Q&A] [answered]
Hey, because I do not create the Items for the Turtles, I don't know how to change the Creative Inventory Tab of my Turtle Upgrades. I already took a look at the Source Code, but I didn't find any solution.

**Comment by SquidDev** [comment, marked as answer]
Easiest solution is to overwrite `fillItemCategory` for your upgrade items to also add the turtles. So something along the lines of:

```java
public void fillItemCategory(@Nonnull ItemGroup group, @Nonnull NonNullList<ItemStack> list) {
  if(!allowededIn(group)) return;
  super.fillItemCategory(group, list);

  // Add turtles
  list.add(makeTurtle(TURTLE_NORMAL, "my_upgrade_id"));
  list.add(makeTurtle(TURTLE_ADVANCED, "my_upgrade_id"));
}

// Probably put this stuff in a helper class:

// Think this is the syntax for ObjectHolder??
@ObjectHolder("computercraft:turtle_normal") public static Item TURTLE_NORMAL;
@ObjectHolder("computercraft:turtle_advanced") public static Item TURTLE_ADVANCED;

public ItemStack makeTurtle(Item turtle, String upgrade) {
  ItemStack stack = new ItemStack(turtle);
  stack.getOrCreateTag().putString(RightUpgrade, upgrade);
  return stack;
}
```

**Reply by SirEndii** [reply]
Thank you! That just works perfectly.
One other question: all the turtles looks the same because I use the turtle_advanced model for the upgrade. Do I need to create individual upgrades for the basic and advanced turtles? Or can I check what kind of turtle that is for `getModel`?
Here's the abstract class which I use to create the turtles: https://github.com/Seniorendi/AdvancedPeripherals/blob/master/src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/turtles/BaseTurtle.java#L34

**Reply by SquidDev** [reply]
You're not meant to return the model of the turtle, the turtle renderer takes care of that already. Instead, you're meant to return a model _for your upgrade_.

For instance, turtle tools (pickaxe, shovel, etc...) use the model of their corresponding item:

https://github.com/SquidDev-CC/CC-Tweaked/blob/e48427dbbcb9c0c2211564733e69dd66135b2b25/src/main/java/dan200/computercraft/shared/turtle/upgrades/TurtleTool.java#L96-L103

Meanwhile, speakers use their own custom model. This is a little more complex, as you also need to load the model and textures on initial game load.

https://github.com/SquidDev-CC/CC-Tweaked/blob/e48427dbbcb9c0c2211564733e69dd66135b2b25/src/main/java/dan200/computercraft/shared/turtle/upgrades/TurtleSpeaker.java#L76
